src/NServiceBus.Core/Pipeline/ConnectorContextExtensions.cs

```diff
@@ -61,6 +61,14 @@ public static IIncomingPhysicalMessageContext CreateIncomingPhysicalMessageConte
         }
 
         /// <summary>
+        /// Creates a <see cref="IIncomingPhysicalMessageContext"/> based on the current context.
+        /// </summary>
+        public static IIncomingPhysicalMessageContext CreateIncomingPhysicalMessageContext(this StageConnector<ITransportReceiveContext, IIncomingPhysicalMessageContext> stageConnector, IncomingMessage incomingMessage, ITransportReceiveContext sourceContext)
```

Comment:
@simoncropp and @andreasohlund stated that we made the promise that things
in the pipeline are replaceable with no exceptions. Therefore this has to
be replaceable as well. Unless we backtrack from that decision.